File: ruby/replace_digits.rb
s = 'a1c1e1'
s2 = 'a1b2c3d4e'
s3 = 'v0g6s4d'

def replace_digits(string)
  alphabet = ('a'..'z').to_a

  string_arr = string.split('')

  string_arr.length.times do |i|
    if string_arr[i].to_i && i.odd?
      letter_i = alphabet.index(string_arr[i - 1]) + string_arr[i].to_i
      string_arr[i] = alphabet[letter_i]
    end
  end

  string_arr.join('')
end

p replace_digits(s)
p replace_digits(s2)
p replace_digits(s3)

File: python/replace_digits.py
import string

s = 'a1c1e1'
s2 = 'a1b2c3d4e'
s3 = 'v0g6s4d'

def replaceDigits(s):
  alphabet = list(string.ascii_lowercase)
  stringArr = list(s)

  for i in range(len(stringArr)):
    if stringArr[i].isdigit() and i % 2 == 1:
      letterI = alphabet.index(stringArr[i - 1]) + int(stringArr[i])
      stringArr[i] = alphabet[letterI]

  return ''.join(stringArr)

print(replaceDigits(s))
print(replaceDigits(s2))
print(replaceDigits(s3))